spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -218,27 +227,45 @@ private SpannerStatementQueryExecutor() {
 				keyParts.add(parentKeyParts.next());
 				tagNum++;
 			}
-			orJoiner.add(andJoiner.toString());
+			or.add(andJoiner.toString());
 		}
-		String cond = orJoiner.toString();
+		String whereCause = Optional.ofNullable(persistentEntity.findAnnotation(Where.class))
+				.map(Where::value).orElse("");
+		String condition = join(or.size() == 0 ? "" : or.size() == 1 ? or.get(0)
+				: or.stream().collect(Collectors.joining(") OR (", "(", ")")), whereCause);
```

Comment:
I did it as a first attempt. The stream always returns some value and when we have only one element it became wrapped by bracket - it looks dummy. We need a custom "Joiner" to prevent extra brackets when the stream has a single entry.

Reply:
That's ok if the only element it's wrapped in parentheses. I'm ok with that if that's the price to pay for more readable code :)

```
  static String orJoin(List<String> parts){
    return parts.stream().map(s -> "("+s+")").collect(Collectors.joining(" OR "));
  }
.....
  System.out.println(orJoin(Arrays.asList()));
  System.out.println(orJoin(Arrays.asList("one")));
  System.out.println(orJoin(Arrays.asList("one", "two")));
```
gives this:
```

(one)
(one) OR (two)
```

It looks totally acceptable to me.

Reply:
it's better to use `Collectors.joining(") OR (", "(", ")")` instead of `map(s -> "("+s+")")`.

Ok, I'll follow your advice but I'll update some tests to satisfy additional brackets - I tried not to update any test that not related directly to my change